sdk/dotnet/Pulumi/Serialization/OutputCompletionSource.cs

```diff
@@ -100,5 +99,10 @@ internal static class OutputCompletionSource
             Log.Debug("Fields to assign: " + JsonSerializer.Serialize(result.Keys), resource);
             return result.ToImmutable();
         }
+
+        private static bool IsOutputType(this Type propType)
+        {
+            return propType.IsConstructedGenericType && propType.IsGenericType && propType.GetGenericTypeDefinition() == typeof(Output<>);
```

Comment:
do we need both the `IsConstructedGenericType ` and `IsgenericType` checks?

Reply:
yes, I've hit a problem with the old code which I'll now try to recall...

Reply:
no, we don't :) I reverted the change except I fixed the original conditional